core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -61,32 +65,99 @@
  * public enum Color {
  *     RED, GREEN, BLUE;
  * }
- * public static final Setting<Color> MY_BOOLEAN = new Setting<>("my.color.setting", Color.RED.toString(), Color::valueOf, false, Scope.CLUSTER);
+ * public static final Setting<Color> MY_BOOLEAN =
+ *     new Setting<>("my.color.setting", Color.RED.toString(), Color::valueOf, SettingsProperty.ClusterScope);
  * }
  * </pre>
  */
 public class Setting<T> extends ToXContentToBytes {
+
+    public enum SettingsProperty {
+        /**
+         * should be filtered in some api (mask password/credentials)
+         */
+        Filtered,
+
+        /**
+         * iff this setting can be dynamically updateable
+         */
+        Dynamic,
+
+        /**
+         * mark this setting as deprecated
+         */
+        Deprecated,
+
+        /**
+         * Cluster scope.
+         * @See IndexScope
+         * @See NodeScope
+         */
+        ClusterScope,
```

Comment:
I think we should just have 2 scopes, `Node` and `Index` and if a `Node` scope setting also `Dynamic` then it's basically a cluster scope setting all the others are really just node settings no?